### PR TITLE
[video_player] added iOS exception on incorrect asset path

### PR DESCRIPTION
**Pull request by zopagaduanjr**
surfaced the incorrect asset path error to dart using try catch.

Fixes [#118660](https://github.com/flutter/flutter/issues/118660)

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

File: packages/video_player/video_player_avfoundation/CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,7 @@
-## NEXT
+## 2.4.7
 
 * Updates minimum supported SDK version to Flutter 3.3/Dart 2.18.
+* Added iOS exception on incorrect asset path
```
Comment (hellohuanlin) [inline packages/video_player/video_player_avfoundation/CHANGELOG.md:4]: Adds

File: packages/video_player/video_player_avfoundation/ios/Classes/FLTVideoPlayerPlugin.m [hunk 2]
```diff
@@ -620,10 +620,16 @@ - (FLTTextureMessage *)create:(FLTCreateMessage *)input error:(FlutterError **)e
...
-                                      frameUpdater:frameUpdater
-                                     playerFactory:_playerFactory];
-    return [self onPlayerSetup:player frameUpdater:frameUpdater];
+    @try {
+        player = [[FLTVideoPlayer alloc] initWithAsset:assetPath
+                                          frameUpdater:frameUpdater
+                                         playerFactory:_playerFactory];
+        return [self onPlayerSetup:player frameUpdater:frameUpdater];
+    }
+    @catch (NSException *exception) {
```
Comment (hellohuanlin) [inline packages/video_player/video_player_avfoundation/ios/Classes/FLTVideoPlayerPlugin.m:629]: did it crash before when the exception is thrown?

**Review by hellohuanlin — COMMENTED** [review]
Can you add some description with the exception logs before and after your change.

**Comment by zopagaduanjr** [comment]
current exception logs when asset path is incorrect

```
Lost connection to device.
Exception: Unable to terminate com.example.delete on 3665098C-8CA5-41DD-9915-520BC16A4457:
ProcessException: Process exited abnormally:
An error was encountered processing the command (domain=NSPOSIXErrorDomain, code=3):
Simulator device returned an error for the requested operation.
found nothing to terminate
Underlying error (domain=NSPOSIXErrorDomain, code=3):
	The operation couldn’t be completed. found nothing to terminate
	found nothing to terminate
  Command: /usr/bin/arch -arm64e xcrun simctl terminate 3665098C-8CA5-41DD-9915-520BC16A4457 com.example.delete
Failed to send request: {"jsonrpc":"2.0","id":"21","method":"getObject","params":{"isolateId":"isolates/842831092504847","objectId":"libraries/@609171358"}}

```

**Comment by zopagaduanjr** [comment]
the proposed try catch update will now raise a platformException, and will not instantly crash the app

```
[VERBOSE-2:dart_vm_initializer.cc(41)] Unhandled Exception: PlatformException(video_player, *** -[NSURL initFileURLWithPath:]: nil string parameter, null, null)
#0      AVFoundationVideoPlayerApi.create (package:video_player_avfoundation/src/messages.g.dart:298:7)
<asynchronous suspension>
#1      AVFoundationVideoPlayer.create (package:video_player_avfoundation/src/avfoundation_video_player.dart:65:37)
<asynchronous suspension>
#2      VideoPlayerController.initialize (package:video_player/video_player.dart:404:19)
<asynchronous suspension>
```

File: packages/video_player/video_player_avfoundation/ios/Classes/FLTVideoPlayerPlugin.m (hunk 2 above)
Comment (zopagaduanjr) [reply packages/video_player/video_player_avfoundation/ios/Classes/FLTVideoPlayerPlugin.m:629]: for ios yes. using the code sample of https://github.com/flutter/flutter/issues/118660#issue-1537405692, it should crash and look like this

Screenshot 2023-06-28 at 9 47 56 PM

**Review by hellohuanlin — COMMENTED** [review]
looks like a good change. can you add some tests?

File: packages/video_player/video_player_avfoundation/test/avfoundation_video_player_test.dart [hunk 3]
```diff
@@ -135,6 +135,15 @@ void main() {
...
+    test('create with incorrect asset', () async {
+      try {
+        await player.create(DataSource(sourceType: DataSourceType.asset));
+        fail('should throw PlatformException');
+      } catch (e) {
+        expect(e, isException);
+      }
+    });
+
     test('create with network', () async {
```
Comment (zopagaduanjr) [inline packages/video_player/video_player_avfoundation/test/avfoundation_video_player_test.dart:147]: is this a valid test?

Comment (hellohuanlin) [reply packages/video_player/video_player_avfoundation/test/avfoundation_video_player_test.dart:147]: where's the asset path here? (I assume you wanna pass in an invalid url or something?)

Comment (zopagaduanjr) [reply packages/video_player/video_player_avfoundation/test/avfoundation_video_player_test.dart:147]: yes I want to input an invalid asset path, hence I put null. or should I be explicit and add an invalid asset path in String?

Comment (hellohuanlin) [reply packages/video_player/video_player_avfoundation/test/avfoundation_video_player_test.dart:147]: yeah that'd be easier to read

File: packages/video_player/video_player_avfoundation/test/avfoundation_video_player_test.dart [hunk 4]
```diff
@@ -135,6 +135,18 @@ void main() {
       expect(textureId, 3);
     });
 
+    test('create with incorrect asset', () async {
+      try {
+        await player.create(DataSource(
+          sourceType: DataSourceType.asset,
+          asset: 'incorrectAsset',
```
Comment (hellohuanlin) [inline packages/video_player/video_player_avfoundation/test/avfoundation_video_player_test.dart:142]: nit: can you make it look more like a path, e.g. `/path/to/incorrect_asset`

**Review by hellohuanlin — APPROVED** [review]

**Review by tarrinneal — CHANGES_REQUESTED** [review]
You'll need to run the format command `dart pub global run flutter_plugin_tools format --packages video_player` before this can land.

**Comment by zopagaduanjr** [comment]
finished running the format command

File: packages/video_player/video_player_avfoundation/test/avfoundation_video_player_test.dart [hunk 5]
```diff
@@ -135,6 +135,18 @@ void main() {
       expect(textureId, 3);
     });
 
+    test('create with incorrect asset', () async {
```
Comment (tarrinneal) [inline packages/video_player/video_player_avfoundation/test/avfoundation_video_player_test.dart:138]: I'd rename this to something like 'create with incorrect asset throws exception' or something like that.

**Review by tarrinneal — CHANGES_REQUESTED** [review]
lgtm with one small change.

**Review by tarrinneal — APPROVED** [review]

**Pull request merged**